spec/checks_collection_spec.rb

```diff
@@ -0,0 +1,60 @@
+require_relative 'spec_helper'
+
+describe RepoAudit::ChecksCollection do
+  let(:check1) { spy('check1') }
+  let(:check2) { spy('check2') }
+  let(:checks_config) { [check1, check2] }
+
+  subject { described_class.new(config: checks_config) }
+
+  describe '.new' do
+    before do
+      allow(RepoAudit::ChecksFactory).to receive(:build)
+    end
+
+    it 'builds the checks using their configuration' do
+      expect(subject).to be_an_instance_of(described_class)
+
+      expect(check1).to have_received(:type)
+      expect(check1).to have_received(:metadata)
+      expect(check1).to have_received(:arguments)
+
+      expect(check2).to have_received(:type)
+      expect(check2).to have_received(:metadata)
+      expect(check2).to have_received(:arguments)
+    end
+
+    it 'behaves like an array' do
+      expect(subject.size).to eq(2)
+      expect(subject.each).to be_an_instance_of(Enumerator)
+    end
+  end
+
+  describe '#run' do
+    let(:repo) { double('repository') }
+    let(:check) { spy('check') }
+
+    before do
+      allow(RepoAudit::ChecksFactory).to receive(:build).and_return(check)
+    end
+
+    it 'should receive the repository as argument' do
+      expect { subject.run }.to raise_error(ArgumentError)
+    end
+
+    it 'iterates through the checks, invoking their `#run` method' do
+      subject.run(repo)
+      expect(check).to have_received(:run).twice.with(repo)
```

Comment:
In this specific scenario it is because the collection contains 2 checks, and thus `run` must be called twice, to ensure it iterates through them.

But I can see where the confusion comes from, because the 2 checks in the collection are the same double, so it looks like the same check receives `run` twice.

I will rewrite this test to ensure we have two different doubles in the collection and expect `run` to be called once in each of them.